runtime/src/asset.rs

```diff
@@ -961,18 +963,22 @@ decl_module! {
             ticker.canonize();
             ensure!(Self::is_owner(&ticker, did), "user is not authorized");
 
-            <Documents<T>>::insert((ticker, name), (uri, document_hash, <pallet_timestamp::Module<T>>::get()));
-            Ok(())
+            for doc in documents.iter() {
+                Self::add_document(ticker, doc.clone());
```

Comment:
we don't need to clone `doc` since we are not using it anywhere else

Reply:
clone was needed because of "cannot move out of" errors in for loop. In any case, I used `into_iter` which does the move implicitly.